core/src/main/java/com/sequenceiq/distrox/v1/distrox/controller/DistroXUpgradeV1Controller.java

```diff
@@ -80,6 +84,23 @@ public DistroXUpgradeV1Response upgradeClusterByCrnInternal(@ResourceCrn String
                 dataHubOsUpgradeEntitled));
     }
 
+    @Override
+    @CheckPermissionByResourceName(action = AuthorizationResourceAction.UPGRADE_DATAHUB)
```

Comment:
why do you tie to upgrade here? Isn't it meant to be a general endpoint in the end? I would rather place it in the general controller and maybe introduce a new action, as I couldn't find an appropriate for the first sight

Reply:
I can, but this action should mostly be used when an upgrade fails. That is, only persons who are eligible for upgrade should be using it.